### PR TITLE
Fix for issue #856

### PR DESCRIPTION
**Pull request by MeanSquaredError**
This PR fixes issue #856
The problem was caused by the fact that when searching for an existing .bbl file we were passing a relative pathname (actually just a .bbl filename) for that existing .bbl file. Since the user also specified relative pathname for the additional log files, the generic search function `BuildManager::findfile()` failed to find the .bbl file.

The proposed fix replaces `BuildManager::findfile()` with a separate class `FindInDirs`. That class has the following features:

1. It does everything that `BuildManager::findfile()` could do
2. It enforces search in absolute directory pathnames. This enforcing is implemented by specifying an absolute directory in the constructor ( `resolveDir`). Any search directory that is loaded afterwards is checked and if it is not absolute, then it is resolved to an absolute directory by prepending `resolveDir` to it. This enforcing of absolute pathnames takes care of the problem with the lookup of .bbl filenames.
3. The return value from `FindInDirs::findAbsolute()` is always absolute. On the other hand `BuildManager::findfile()` could return a relative pathname if it was passed a relative pathname and that relative pathname was found relative to current directory.

Ideally `FindInDirs` will be used to replace all the other search functions like `findResourceFile()`, etc. but it probably needs to be extended with extra properties/methods to achieve that.

### CODE REVIEW

**Comment by sunderme** [comment]
I have two remarks
1) i would like to see documentation/comments, especially on code which is used all ober the place
doxygen comments arenpreffered and supported in qtcreator. Try typing /*! above any function declaration
2) the check for existence and readability could be moved into that class as well.

File: src/findindirs.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	loadDirs(splitPaths(dirs));
+}
+
+void FindInDirs::loadDirs(const QStringList &dirs)
+{
+	foreach(const QString &oneDir, dirs) {
+		loadOneDir(oneDir);
+	}
+}
+
```
Comment (sunderme) [inline src/findindirs.cpp]: not sure that we need that extra function.
loadDirs should be sufficient

**Comment by MeanSquaredError** [comment]
OK, I did the following changes:

1. Added Doxygen documentation for the `FindInDirs` class.
2. Added an extra argument to `FindInDirs::FindInDirs` that allows check for file readability.
3. Removed `FindInDirs::loadOneDir`

**Comment by sunderme** [comment]
perfect

**Pull request merged**